packages/arb-avm-cpp/cavm/carbcore.cpp

```diff
@@ -551,21 +551,28 @@ CMachine* arbCoreTakeMachine(CArbCore* arbcore_ptr,
         arbCore->takeExecutionCursorMachine(*executionCursor).release());
 }
 
-CMachine* arbCoreGetMachineForSideload(CArbCore* arbcore_ptr,
-                                       uint64_t block_number) {
+CMachineResult arbCoreGetMachineForSideload(CArbCore* arbcore_ptr,
+                                            uint64_t block_number,
+                                            int allow_slow_lookup) {
     auto arbcore = static_cast<ArbCore*>(arbcore_ptr);
 
     try {
-        auto machine = arbcore->getMachineForSideload(block_number);
+        auto machine =
+            arbcore->getMachineForSideload(block_number, allow_slow_lookup);
         if (!machine.status.ok()) {
             std::cerr << "Failed to load machine for sideload "
                       << machine.status.ToString() << std::endl;
-            return nullptr;
+            return {nullptr, 0};
+        }
+
+        if (!machine.data) {
+            // Machine not found in memory cache and database lookup disabled
```

Comment:
If the machine isn't found, the status check above with fail since we return `NotFound` in that case. I think we want to special case check for `NotFound` in the case where `allow_slow_lookup` is true

Reply:
:+1: